### PR TITLE
Add GUI documentation

### PR DESCRIPTION
**Pull request by gassmoeller**
This PR adds some documentation to the new GUI option in the manual and in changes.h. When this has been merged I will send an additional email on the mailing list to make sure people notice the cool new features :smile:.

### CODE REVIEW

File: doc/manual/manual.tex [hunk 1]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
 the syntax understood, reference the documentation of the muparser library
 linked to above.
 
+  
+\subsection{A graphical user interface for \aspect{}}
```
Comment (bangerth) [inline doc/manual/manual.tex]: Maybe say "A graphical user interface for editing \aspect{} parameter files"?

File: doc/manual/manual.tex [hunk 2]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
...
+Preparing a parameter file in a text editor can be a tedious task, not
+only because the number of input parameters has grown considerably during the
+development of \aspect{}, but also because remembering the names of commonly
+used options is a waste of (human) memory. Therefore, we provide a graphical
```
Comment (bangerth) [inline doc/manual/manual.tex]: :-)

File: doc/manual/manual.tex [hunk 3]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
...
+user interface that builds upon an available program for deal.II. This GUI
```
Comment (bangerth) [inline doc/manual/manual.tex]: don't we use `\dealii{}`? same below

File: doc/manual/manual.tex [hunk 4]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
...
+subsections we describe installing and using this user interface. 
+
+\subsubsection{Installing parameter-GUI}
+The deal.II parameter-GUI program can be downloaded at \url{https://github.com/dealii/parameter_gui},
+and is compiled using the \texttt{cmake} program just like \aspect{} itself. The program has no
+dependencies except for the Qt development libraries that should be available as packages for 
+most Linux distributions and can also be obtained for all major operating systems at
+\url{https://www.qt.io/download-open-source/}.
+
+Example steps for installing the parameter-gui:
```
Comment (bangerth) [inline doc/manual/manual.tex]: use a complete sentence

File: doc/manual/manual.tex [hunk 5]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
...
+When configuring \aspect{} after executing the above steps, it should automatically pick up the
+location of the parameter-GUI, and will create a new script named \texttt{aspect-gui} within the build folder.
+If this does not happen, it is possible to hand over the location of the parameter-GUI as a cmake
+variable during configuration (e.g. \texttt{cmake -D PARAMETER\_GUI\_EXECUTABLE=path\_to\_your\_executable}).
+
+The aspect-gui script can be executed from any folder either with no argument or with one argument that contains the path to
+an existing parameter file. The script will run \aspect{} with the given (or an empty) parameter file, generate a database
+of existing input parameters, and open the parameter-GUI program with this database. The resulting window looks similar to Fig.~\ref{fig:aspect-gui}. If an existing parameter file was
+given, all parameter fields are pre-filled with the values set in the file instead of the default values.
+In the appearing window you can change parameters as necessary, and then save the file as a new parameter file. This parameter file can then be used
```
Comment (bangerth) [inline doc/manual/manual.tex]: -> In the program's main window

File: doc/manual/manual.tex [hunk 6]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
...
+to start an \aspect{} model as usual. Note that it is possible to prepare and execute the parameter file with different versions of \aspect{}, e.g. if you prepare parameter files on a local machine, and execute the model on a remote compute cluster.
+Note however that if the two \aspect{} versions contain different default values or parameter names have changed this can lead
```
Comment (bangerth) [inline doc/manual/manual.tex]: add comma after "changed" to make it easier to read

File: doc/manual/manual.tex [hunk 7]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
...
+to unexpected model behaviour or even unusable parameter files.
```
Comment (bangerth) [inline doc/manual/manual.tex]: behavior

File: doc/manual/manual.tex [hunk 8]
```diff
@@ -3912,7 +3912,59 @@ \subsubsection{A note on the syntax of formulas in input files}
...
 
+\begin{figure}
+\hfill
+\includegraphics[width=0.4\textwidth]{aspect-gui.png}
+\hfill
+\phantom.
+\caption{\it The parameter GUI lists all available parameter options, and allows to change and save them into a new parameter file. Input fields know about the type of the variable and will display useful input fields to change them (e.g. drop-down menus, file dialogs, text fields).}
```
Comment (bangerth) [inline doc/manual/manual.tex]: can you re-word the second use of "input fields"?

**Comment by tjhei** [comment]
awesome, thanks! :+1:

**Pull request merged**